functions/slack/main.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018, Google, LLC.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START functions_slack_setup]
+import json
+
+import apiclient
+from flask import jsonify
+
+with open('config.json', 'r') as f:
+    data = f.read()
+config = json.loads(data)
+
+kgsearch = apiclient.discovery.build('kgsearch', 'v1',
```

Comment:
We have to use apiclient here? What's this for? I'm not familiar with "kgsearch"

Reply:
It's the Knowledge Graph API, which is used in the Slack tutorial to return search results. The API is not a part of GCP products and doesn't have veneer client library :(